docs/machine-learning/Naive_Bayes.md

```diff
@@ -0,0 +1,159 @@
+   
+2. **Calculate Prior Probabilities**:  
+   The prior probability of each class is computed based on the frequency of each class in the training data.
+
+3. **Calculate Likelihood**:  
+   For each feature and class, the likelihood is calculated by determining how likely it is to observe a particular feature value given the class.
+
+4. **Apply Bayes' Theorem**:  
+   Using Bayes' Theorem, the posterior probability of each class is calculated based on the priors and likelihoods:
+   
+  ![image](https://github.com/user-attachments/assets/b7c360d9-7fd6-45d6-ba13-8be3617da7ef)
+
+   
+   where P(y|X) is the posterior probability of the class given the feature vector, P(X|y) is the likelihood, P(y) is the prior, and P(X) is the evidence.
+
+5. **Classify New Data**:  
+   For a new data point, the algorithm computes the posterior probability for each class and assigns the label of the class with the highest probability.
+
+### Problem Statement:
+Given a dataset with multiple features and corresponding class labels, the objective is to train a Naive Bayes classifier that can predict the class label for new, unseen data based on the calculated probabilities.
+
+### Key Concepts:
+- **Bayes' Theorem**:  
+  Bayes' Theorem is a mathematical formula used to calculate conditional probabilities. It is expressed as:
+  
+![image](https://github.com/user-attachments/assets/a54ebaeb-7d44-4019-909f-de463d0b7e73)
+
+  
+  where:
+  -  P(A|B) is the posterior probability of event A given that B is true.
+  -  P(B|A) is the likelihood of observing B given that A is true.
+  -  P(A) is the prior probability of event A.
+  -  P(B) is the total probability of event B.
+
+- **Prior Probability**:  
+  The prior probability represents the initial belief about the class labels before considering any feature values, based on the distribution of classes in the training data.
+  
+- **Likelihood**:  
+  The likelihood is the probability of observing the feature values given a specific class label.
+
+- **Posterior Probability**:  
+  The posterior probability is the probability of a class label after considering the observed feature values, which is what Naive Bayes uses to make predictions.
+
+- **Naive Assumption**:  
+  Naive Bayes assumes that all features are independent given the class, simplifying the probability calculations. This assumption, although unrealistic in many cases, allows for efficient computation and often leads to good results.
+
+### Split Criteria:
+Naive Bayes splits data based on the **highest posterior probability** for each class, assigning the class label that maximizes this probability.
+
+### Time Complexity:
+- **Training Complexity**:  
+  Training involves calculating probabilities for each feature and class, resulting in a time complexity of \( O(n \cdot k) \), where \( n \) is the number of features and \( k \) is the number of classes.
```

Comment:
replace `\( O(n \cdot k) \)` with `$O(n \cdot k)$` and `\( n \)` with `$n$` and also, `\( k \)` with `$k$`